opslevel/resources_opslevel_team_tag.go

```diff
@@ -1,159 +1,255 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Tag key names are stored in OpsLevel as lowercase so we need to ensure the configuration is written as lowercase
+var (
+	TagKeyRegex    = regexp.MustCompile(`\A[a-z][0-9a-z_\.\/\\-]*\z`)
+	TagKeyErrorMsg = "tag key name must start with a letter and be only lowercase alphanumerics, underscores, hyphens, periods, and slashes."
+)
+
+var _ resource.ResourceWithConfigure = &TeamTagResource{}
+
+var _ resource.ResourceWithImportState = &TeamTagResource{}
+
+type TeamTagResource struct {
+	CommonResourceClient
+}
+
+func NewTeamTagResource() resource.Resource {
+	return &TeamTagResource{}
+}
+
+type TeamTagResourceModel struct {
+	Id          types.String `tfsdk:"id"`
+	Key         types.String `tfsdk:"key"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+	Team        types.String `tfsdk:"team"`
+	TeamAlias   types.String `tfsdk:"team_alias"`
+	Value       types.String `tfsdk:"value"`
+}
+
+func NewTeamTagResourceModel(teamTag opslevel.Tag) TeamTagResourceModel {
+	teamResourceModel := TeamTagResourceModel{
+		Key:   RequiredStringValue(teamTag.Key),
+		Value: RequiredStringValue(teamTag.Value),
+		Id:    ComputedStringValue(string(teamTag.Id)),
+	}
+
+	return teamResourceModel
+}
+
+func (teamTagResource *TeamTagResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team_tag"
+}
+
+func (teamTagResource *TeamTagResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Team Tag Resource",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of this resource.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_updated": schema.StringAttribute{
+				Computed: true,
+			},
+			"key": schema.StringAttribute{
+				Description: "The tag's key.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(TagKeyRegex, TagKeyErrorMsg),
+				},
+			},
+			"value": schema.StringAttribute{
+				Description: "The tag's value.",
+				Required:    true,
+			},
+			"team": schema.StringAttribute{
+				Description: "The id of the team that this will be added to.",
+				Optional:    true,
+				Validators: []validator.String{
+					IdStringValidator(),
+					stringvalidator.AtLeastOneOf(path.MatchRoot("team"),
+						path.MatchRoot("team_alias")),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"team_alias": schema.StringAttribute{
+				Description: "The alias of the team that this will be added to.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
```

Comment:
TODO: `team_alias` also needs the validator chunk from `team`
```go
Validators: []validator.String{
					IdStringValidator(),
					stringvalidator.AtLeastOneOf(
						path.MatchRoot("team"),
						path.MatchRoot("team_alias")),
				},
```

Reply:
@davidbloss to my surprise, the examples actually show it requires it to only be set on one of the fields.

![image](https://github.com/OpsLevel/terraform-provider-opslevel/assets/37668804/582ac3f6-2038-479b-941a-6c805ce83f64)